Sources/DesignSystem/Dimensions/DimensionFileGenerator.swift

```diff
@@ -0,0 +1,28 @@
+import Foundation
+
+/// Dimensions file generator
+final class DimensionFileGenerator {
+    // MARK: - Static internal functions
+    static func generate(from dimensions: [DimensionDataModel], at path: String) {
```

Comment:
maybe create a `DimensionType` enum and use a dictionary instead of an array: `[DimensionType: DimensionDataModel]`, with that you will be able to create on file/enum per type to keep readability and have more smallest files at the end. 

```
enum DimensionType {
        case gap
        case padding
        ...
        
        var className: String {
            switch self {
            case .gap:
                return "DimensionGap" // Maybe found more relevant name
            case .padding:
                return "DimensionPadding" // Maybe found more relevant name
            case ...:
                ...
            }
        }
        
        var path: String {
            return "/" + self.className + ".swift"
        }
    }
```
```
static func generate(from dimensions: [DimensionType: DimensionDataModel], at path: String) {
...
dimensions.keys.forEach { type in
            var contentFileStr: String = "import Foundation"
            + String(repeating: "\n", count: 2)
            + "public enum "
            + type.className
            + "{"
            
            dimensions[type]?.forEach({ dimension in
                contentFileStr += "\n"
                + "\t"
                + "public static let \($0.name): CGFloat = \($0.value)"
            })
                    
            contentFileStr += "\n"
            + "}"
            
            do {
                try contentFileStr.write(
                    to: URL(fileURLWithPath: path + type.path),
                    atomically: true,
                    encoding: .utf8
                )
            } catch {
                print(error.localizedDescription)
            }
        }
...
}
```